plugin-scaffolder: VirtualizedListbox tests and padding

danny-may opened the pull request:
## Hey, I just made a Pull Request!

This PR adds tests for the VirtualizedListbox and adds back in the half element padding to the element dropdown that was removed

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Rugvip commented on .changeset/unlucky-cherries-rush.md:
```diff
@@ -0,0 +1,5 @@
+'@backstage/plugin-scaffolder': patch
+---
+
+Implement forwarding of additional props from the EntityPicker autocomplete to the VirtualizedListbox outer element. This fixes an accessibility issue where you are unable to use the arrow keys to navigate the dropdown
```
```suggestion
Fixed an accessibility issue where you were unable to use the arrow keys to navigate the `EntityPicker` field dropdown, and other fields that are derived from it.
```

Tweaking it to be a bit more user facing and skip the implementation details.

Rugvip commented on plugins/scaffolder/src/components/fields/VirtualizedListbox.tsx:
```diff
@@ -17,34 +17,46 @@
 
-  const itemsToShow = Math.min(10, itemCount);
-  const height = Math.max(itemSize, itemsToShow * itemSize - 0.5 * itemSize);
+  const height = itemSize * (Math.min(10, itemCount) + 0.5);
```
How come you dropped the `.max` here? I believe that was in place to ensure that an small box is rendered even if the list is empty.

danny-may replied:
Even if itemcount is 0, it will still render a half element size box, so there will always be a box there. I guess I didnt really need to change these lines for this ticket, it had just bugged me that if there were less than 10 items available, the dropdown would always cut the last item in half so I changed it to add 0.5 of an item rather than subtracting it, which meant you couldnt get a 0 size box any more.

Rugvip replied:
Ah yep that works, thank you! :+1:

Rugvip reviewed (changes requested):
Nice, thank you! 👍 

Looks good, just a few nits

benjdlambert commented:
@danny-may sorry yeah I noticed there was some overlap! I thought these were the same PR 😅 , please yes we'd love to have the tests still :pray:

danny-may commented:
@Rugvip @benjdlambert Ive updated the PR to be just the tests I added. Looks like the half element padding got lost in the other PR, so I added it back in here

Rugvip reviewed (approved):
Thank you! 👍 

Let's :shipit:
